orchestrator/graphql/loaders/subscriptions.py

```diff
@@ -1,224 +1,37 @@
-        .join(ResourceTypeTable)
-        .filter(ResourceTypeTable.resource_type.in_(RELATION_RESOURCE_TYPES))
-        .filter(depends_on_instance_values.value.in_(unique_subscription_ids))
-        .filter(in_use_by_subscriptions.status.in_(filter_statuses))
-    )
-
-    return _get_resource_type_relations(query_get_in_use_by_ids)
-
-
-async def _get_depends_on_resource_type_relations(
-    subscription_ids: list[UUID], filter_statuses: list[str]
-) -> list[Relation]:
-    """Get depends_on relations through resource types."""
-    logger.warning("Using legacy RELATION_RESOURCE_TYPES to find depends_on subs")
-
-    depends_on_subscriptions = aliased(SubscriptionTable)
-    in_use_by_instances = aliased(SubscriptionInstanceTable)
-    in_use_by_instance_values = aliased(SubscriptionInstanceValueTable)
-
-    unique_subscription_ids = set(subscription_ids)
-
-    query_get_depends_on_ids = (
-        select(in_use_by_instance_values.value, in_use_by_instances.subscription_id)
-        .select_from(in_use_by_instance_values)
-        .join(in_use_by_instances)
-        .join(
-            depends_on_subscriptions,
-            in_use_by_instance_values.value == sa_cast(depends_on_subscriptions.subscription_id, SaText),
-        )
-        .join(ResourceTypeTable)
-        .filter(ResourceTypeTable.resource_type.in_(RELATION_RESOURCE_TYPES))
-        .filter(in_use_by_instances.subscription_id.in_(unique_subscription_ids))
-        .filter(depends_on_subscriptions.status.in_(filter_statuses))
-    )
-
-    return _get_resource_type_relations(query_get_depends_on_ids)
-
-
-async def _get_in_use_by_relations(subscription_ids: list[UUID], filter_statuses: list[str]) -> list[Relation]:
-    if RELATION_RESOURCE_TYPES:
-        # Find relations through resource types
-        resource_type_relations = await _get_in_use_by_resource_type_relations(subscription_ids, filter_statuses)
-    else:
-        resource_type_relations = []
-    # Find relations through instance hierarchy
-    instance_relations = await _get_in_use_by_instance_relations(subscription_ids, filter_statuses)
-    return list(chain(resource_type_relations, instance_relations))
-
-
-async def _get_depends_on_relations(subscription_ids: list[UUID], filter_statuses: list[str]) -> list[Relation]:
-    if RELATION_RESOURCE_TYPES:
-        # Find relations through resource types
-        resource_type_relations = await _get_depends_on_resource_type_relations(subscription_ids, filter_statuses)
-    else:
-        resource_type_relations = []
-    # Find relations through instance hierarchy
-    instance_relations = await _get_depends_on_instance_relations(subscription_ids, filter_statuses)
-    return list(chain(resource_type_relations, instance_relations))
-
-
-async def in_use_by_subs_loader(keys: list[tuple[UUID, list[str] | None]]) -> list[list[SubscriptionTable]]:
+async def in_use_by_subs_loader(keys: list[tuple[UUID, tuple[str, ...]]]) -> list[list[SubscriptionTable]]:
```

Comment:
If I'm not mistaken the second param can also be None
```suggestion
async def in_use_by_subs_loader(keys: list[tuple[UUID, tuple[str, ...] | None]]) -> list[list[SubscriptionTable]]:
```